packages/core/test/trader/uniswap-broker/UniswapBroker.js

```diff
@@ -0,0 +1,240 @@
+    await pair.sync({ from: deployer });
+    assert.equal(await getPoolSpotPrice(), "1000.0000"); // price should be exactly 1000 TokenA/TokenB.
+  });
+
+  it("Broker can correctly trade the price up to a desired price", async function() {
+    // Say that someone comes and trades in size against the pool, trading a large amount of tokenB for tokenA,
+    // dropping the token price to something off peg. We will compte that the price changes as expected. Say a trade of
+    // 100000 token B for token A. Based on the pool size, the resultant price will be 980.3252 (see logic below for calc).
+    const tradeSizeInTokenB = toBN(toWei("100000"));
+
+    // Find how many tokens will be outputted for an input size of 100000 token B. We are adding token B, to get A. aToB
+    // is false as we are trading from B -> A.
+    const amountOut = await getAmountOut(tradeSizeInTokenB, false);
+
+    // Find the modified versions of the token reservers. As we are trading token B for token A the A reservers should be
+    // decreased by the amountOut and the token B reservers should be increased by the trade size in token B.
+    const tokenAReserve = (await tokenA.balanceOf(pairAddress)).sub(amountOut);
+    const tokenBReserve = (await tokenB.balanceOf(pairAddress)).add(tradeSizeInTokenB);
+
+    // Compute the expected spot price from the two token reserves.
+    const expectedSpotPrice = Number(fromWei(tokenAReserve.mul(toBN(toWei("1"))).div(tokenBReserve))).toFixed(4);
+
+    // Execute the swap.
+    await router.swapExactTokensForTokens(
+      tradeSizeInTokenB, // amountIn. We are selling tokenB for tokenA, therefore tokenB is "in" and tokenB is "out"
+      0, // amountOutMin
+      [tokenB.address, tokenA.address], // path. We are trading from tokenB to tokenA (selling B for A)
+      trader, // recipient of the trade
+      (await web3.eth.getBlock("latest")).timestamp + 10, // deadline
+      { from: trader }
+    );
+    assert.equal(await getPoolSpotPrice(), expectedSpotPrice);
+    assert.equal(await getPoolSpotPrice(), "980.3252");
+
+    // Now that the token is trading out off peg we can test that the broker can correctly trade it back to parity.
+    // First, lets double check the bot calculates the correct trade size with the computeTradeToMoveMarket method.
+    const tradeToMoveMarket = await uniswapBroker.computeTradeToMoveMarket(
+      "1000", // the true price is represented as truePriceTokenA/truePriceTokenB.
+      "1",
+      await tokenA.balanceOf(pairAddress),
+      await tokenB.balanceOf(pairAddress)
+    );
+
+    // For starters, to move the price UP we need to trade token A into the pool for token B. Therefore the aToB should
+    // be true as the token flow should be A for token B to increase the price.
+    assert.isTrue(tradeToMoveMarket.aToB);
+
+    // Secondly, the resultant price after the trade based off the amountIn should be equal to the desired true price.
```

Comment:
This is a good test to check that the resultant price implied by `getAmountOut` results in the desired true price